crates/biome_js_analyze/src/lint/nursery/no_bitwise.rs

```diff
@@ -0,0 +1,191 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::{
+    JsAssignmentExpression, JsBinaryExpression, JsSyntaxToken, JsUnaryExpression, T,
+};
+use biome_rowan::{AstNode, declare_node_union};
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+declare_lint_rule! {
+    /// Disallow bitwise operators
+    ///
+    /// The use of bitwise operators in JavaScript is very rare and often & or | is simply a mistyped && or ||,
+    /// which will lead to unexpected behavior.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// let x = y | z;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// x |= y;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// let x = y || z;
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// The rule provides the option described below.
+    ///
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "allow": ["&", "|", "^", "~", "<<", ">>", ">>>"],
+    ///         "int32Hint": false,
+    ///     }
+    /// }
+    /// ```
+    /// ### allow
```

Comment:
Note we typically leave out the options from a first implementation, to avoid implementing (and maintaining) options that are hardly ever used by users. But seeing how you already did the work, and they don't seem to be too complex, I won't object to it :)